torch/_decomp/decompositions.py

```diff
@@ -1903,6 +1903,19 @@ def uniform(
     )
 
 
+@register_decomposition(aten.geometric_)
+def geometric_(self, p, generator=None):
+    assert generator is None
+    tiny = torch.finfo(self.dtype).tiny
+    return self.copy_(
+        torch.floor(
+            torch.log(torch.clamp(torch.rand_like(self), min=tiny))
+            / torch.log1p(torch.tensor(-p))
```

Comment:
Actually, if `p` is a number, it's better to do `math.log1p(-p)`.
Also, as in the other function, it may be better to implement the `in-place` version in terms of the out-of-place function.